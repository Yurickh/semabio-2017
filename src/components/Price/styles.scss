@import 'ambience/typography';
@import 'base/responsive-prop';
$price-box-color: #343434;

.price-box {
  font-weight: $weight-bold;

  display: flex;
  align-items: center;

  color: $price-box-color;

  & > .currency {
    letter-spacing: -.44px;

    @include responsive-prop(font-size, 20, 20);
  }

  & > .integer {
    letter-spacing: -.88px;

    @include responsive-prop(font-size, 48, 48);
    @include responsive-prop(line-height, 30, 30);
  }

  & > .decimal {
    align-self: flex-end;

    @include responsive-prop(font-size, 14, 14);
  }
}
