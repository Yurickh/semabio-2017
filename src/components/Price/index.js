import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './styles.css'

class Price extends Component {
  render() {
    const integer = Math.trunc(this.props.children)
    const decimal = Math.round((this.props.children - integer) * 100)

    return (
      <div className="price-box">
        <span className="currency">R$</span>
        <span className="integer">{integer}</span>
        <span className="decimal">,{('0000' + decimal).slice(-2)}</span>
      </div>
    )
  }
}

Price.propTypes = {
  children: PropTypes.number,
}

export default Price
