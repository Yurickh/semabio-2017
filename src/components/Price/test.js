import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Price from '../Price'

describe('Price', () => {
  let button

  beforeEach(() => {
    button = shallow(
      <Price>{ 3.14 }</Price>
    )
  })

  it('should render correctly', () => {
    expect(toJSON(button)).toMatchSnapshot()
  })

  it('pads zeros before decimal', () => {
    button.setProps({ children: 4.05 })
    expect(button.find('.decimal').text()).toBe(',05')
  })

  it('pads zeros after decimal', () => {
    button.setProps({ children: 20.2 })
    expect(button.find('.decimal').text()).toBe(',20')
  })
})
