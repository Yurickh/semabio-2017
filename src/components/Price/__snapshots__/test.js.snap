// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price should render correctly 1`] = `
<div
  className="price-box"
>
  <span
    className="currency"
  >
    R$
  </span>
  <span
    className="integer"
  >
    3
  </span>
  <span
    className="decimal"
  >
    ,
    14
  </span>
</div>
`;
